src/cosmosdb/session/DocumentSession.ts

```diff
@@ -386,14 +513,146 @@ export class DocumentSession {
                 message = `${message}\nActivityId: ${error.ActivityId}`;
             }
 
-            const showLogButton = l10n.t('Go to output');
-            if (await vscode.window.showErrorMessage(message, showLogButton)) {
-                ext.outputChannel.show();
-            }
+            this.showError(message);
+
             throw new Error(`${message}, ${parsedError.message}`);
         } else {
-            await vscode.window.showErrorMessage(message);
+            vscode.window.showErrorMessage(message);
             throw new Error(message);
         }
     }
+
+    private showError(message: string): void {
+        ext.outputChannel.error(message);
+        vscode.window.showErrorMessage(message, l10n.t('Go to output')).then((result) => {
+            if (result) {
+                ext.outputChannel.show();
+            }
+        });
+    }
+
+    private showInfo(message: string): void {
+        ext.outputChannel.appendLog(message);
+        vscode.window.showInformationMessage(message, l10n.t('Go to output')).then((result) => {
+            if (result) {
+                ext.outputChannel.show();
+            }
+        });
+    }
+
+    private async showConfirmation(message: string): Promise<boolean> {
+        const continueItem: vscode.MessageItem = { title: l10n.t('Continue') };
```

Comment:
Shouldn't we use [getConfirmationAsInSettings](https://github.com/microsoft/vscode-cosmosdb/blob/a8ad13c0b5052cbf41bb15a6fcbd265271a83c89/src/utils/dialogs/getConfirmation.ts#L25) here?

Reply:
Done